packages/postgres-database/src/simcore_postgres_database/utils_projects_nodes.py

```diff
@@ -274,8 +278,12 @@ async def get_project_id_from_node_id(
         get_stmt = sqlalchemy.select(projects_nodes.c.project_uuid).where(
             projects_nodes.c.node_id == f"{node_id}"
         )
-        project_id = await connection.scalar(get_stmt)
-        if project_id is None:
+        result = await connection.execute(get_stmt)
+        project_ids = await result.fetchall()
+        if not project_ids:
             msg = f"No project found containing {node_id=}"
             raise ProjectNodesNodeNotFound(msg)
-        return uuid.UUID(project_id)
+        if len(project_ids) > 1:
+            msg = f"Multiple project found containing {node_id=}"
+            raise ProjectNodesNonUniqueNodeFoundError(msg)
```

Comment:
This can occur, but what should it be done if it occurs? Could you propose an action for the user or whoever reads this error in the logs? That would be great.

Reply:
well this could have occured already before. at least now there is an exception raised. in our current production system this never happens so we should be safe. but it is worse if we suddenly start doing different projects with the same node ID inside if we do not realize it. that is the meaning of that.

Reply:
discussed.